### PR TITLE
Taboola Bid Adapter: onBidWon, userSyncs, gpp support and FPD 

### PR DESCRIPTION
**Pull request by ahmadlob**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- supporting and parsing first party data. 
- implementing onBidWon function 
- implementing getUserSyncs function -  image sync

### CODE REVIEW

**Comment by ahmadlob** [comment]
prebid.github.io PR:
https://github.com/prebid/prebid.github.io/pull/4235

File: modules/taboolaBidAdapter.js [hunk 1]
```diff
@@ -149,9 +153,34 @@ export const spec = {
 
     return bidResponses.map((bidResponse) => getBid(bids, currency, bidResponse)).filter(Boolean);
   },
+  onBidWon: (bid) => {
+    if (bid.nurl) {
+      ajax(replaceAuctionPrice(bid.nurl, bid.originalCpm));
+    }
+  },
+  getUserSyncs: function(syncOptions, serverResponses, gdprConsent, uspConsent) {
```
Comment (patmmccann) [inline modules/taboolaBidAdapter.js:161]: Why not pass the gpp string?

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/taboolaBidAdapter.js [hunk 2]
```diff
@@ -149,9 +153,34 @@ export const spec = {
 
     return bidResponses.map((bidResponse) => getBid(bids, currency, bidResponse)).filter(Boolean);
   },
+  onBidWon: (bid) => {
+    if (bid.nurl) {
+      ajax(replaceAuctionPrice(bid.nurl, bid.originalCpm));
```
[the remark this answers is not in the document]
Comment (ahmadlob) [reply modules/taboolaBidAdapter.js]: @patmmccann 
Not sure what I'm missing, the replacing and resolving of the price occurs before the ajax call - I don't pass a callback to the ajax, I call the function and thus it should resolve the bid price firstly. 
I wrote the code more cleaner - will appreciate your feedback.

File: modules/taboolaBidAdapter.js (hunk 1 above)
Comment (ahmadlob) [reply modules/taboolaBidAdapter.js]: @patmmccann 

Didn't know you support gpp and that it's a valid argument for this function (here it is not mentioned at all https://docs.prebid.org/dev-docs/integrate-with-the-prebid-analytics-api.html). 

Now - I pass the gpp string if it exists.

File: modules/taboolaBidAdapter.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/taboolaBidAdapter.js]: Why wouldn't you fill this in server side? Relying on the client to do so subjects you to a trivial attack vector for fraud.

File: modules/taboolaBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/taboolaBidAdapter.js]: it's brand new, nice work

File: modules/taboolaBidAdapter.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (patmmccann) [reply modules/taboolaBidAdapter.js]: I'm merging as this is not a rules violation, just a bad idea.

**Review by patmmccann — APPROVED** [review]

**Pull request merged**